The reason that sunkit-image exists (#200)
sunpy/sunkit-image: docs/index.rst

          generated/gallery/index
          changelog

Mission Statement
=================

> The goal of the ``sunkit-image`` library is to provide access to image processing routines that are:

Author: ```suggestion

The goal of ``sunkit-image`` is not to be a general purpose image processing library.

The goal of ``sunkit-image`` is to provide access to image processing routines that are:
```

1. Focused on being applied to solar physics data.

> 2. Are published in the literature.

Reviewer: What about that are in-prep? I like the idea of people writing their algorithms in sunkit-image before they publish so they can say that's where it will be in the paper? Or does this open us up to "oh we are going to publish this eventually"?

Author: How about if they can prove its in prep?

Reviewer: How about we state it must be in prep and if we are skeptical for any reason we can only merge the PR when it's close to publication?

Author: Sure.

> 2. Are published in the literature.

Author: ```suggestion
2. Are published in the literature or in preparation to be published.
```

> 3. If there is no publication, it must be widely used or perceived as generally useful throughout the solar physics community.

Reviewer: I see the reason for this but it's a bit wolly? :sheep:

Reviewer: Let's keep the wool. :yarn:

   Examples include co-alignment routines that compensate for incorrect pointing, solar feature identification algorithms, and filtering functions. 

As this is a SunPy affiliated package, the inputs to these routines are required to be Maps (sunpy) or NDCubes (ndcube).

If the code is already in a released package, we will wrap calls to the existing package in a way that makes it easy to use with `Map` or `NDCube` objects. Additional modifications to such packages are outside the scope of sunkit-image. 
We will not copy code from other packages into this one.

> Finally, this library is not aiming to be a general purpose image processing library.

Author: ```suggestion
```

          generated/gallery/index
          changelog

Mission Statement
=================

The goal of the ``sunkit-image`` library is to provide access to image processing routines that are:

> 1. Focused on being applied to solar physics data.

Reviewer 2: ```suggestion
1. Focused on being applied to solar image data.
```

We need to say something about what kind of date this package is focused on. It's specifically focused on operations on image data.

Reviewer 2: I'm not sure if we want to more strictly define "image" here

2. Are published in the literature or in preparation to be published.

> 3. If there is no publication, it must be widely used throughout the solar physics community.

Reviewer 2: This last point is, grammatically, unlike the first two.

```suggestion
3. Widely used throughout the solar physics community.
```

>    One example might be co-alignment routines that exist in SSWIDL.

Reviewer: ```suggestion
   One example might be co-alignment routines that exist in SSWIDL.
```
All possible coalignment routines in SSW? Do we need this example?

Author: I didn't have a good example, so I came up with this.

>    One example might be co-alignment routines that exist in SSWIDL.

Reviewer 2: ```suggestion
   Examples include co-alignment routines that compensate for incorrect pointing, solar feature identification algorithms, and filtering functions. 
```

> As this is a SunPy affiliated package, the inputs to these routines are required to be Maps (sunpy) or ANDCubes (ndcube).

Reviewer: ```suggestion
As this is a SunPy affiliated package, the inputs to these routines are required to be Maps (sunpy) or NDCubes (ndcube).
```

> As this is a SunPy affiliated package, the inputs to these routines are required to be Maps (sunpy) or ANDCubes (ndcube).

Reviewer 2: Is this true? We have ones that aren't (eg the time lag method). Should this be moved?

Author: Does it not work on maps?

Reviewer 2: No. Only arrays

> The ownership of the code is also important.

Reviewer 2: ```suggestion
```

> If the code is already in a released package, we will wrap it in a way that makes it easy to use with Maps or ANDCubes but otherwise leave it alone.

Reviewer 2: ```suggestion
If the code is already in a released package, we will wrap calls to the existing package in a way that makes it easy to use with `Map` or `NDCube` objects. Additional modifications to such packages are outside the scope of sunkit-image. 
```

We will not copy code from other packages into this one.

> Finally, this library is not aiming to be a general purpose image processing library.

Reviewer 2: Could we move this to the beginning? Or maybe have a subsection on what sunkit-image is not?

Author: Yeah I can move this to the start.

> 
> If the code is already in a released package, we will wrap calls to the existing package in a way that makes it easy to use with `sunpy.map.Map` or `ndcube.NDCube` objects.
> Additional modifications to such packages are outside the scope of ``sunkit-image``. 
> We will not copy code from other packages into this one.

Author: ```suggestion
If the code is already in a released package, we will wrap calls to the existing package in a way that makes it easy to use with `sunpy.map.Map` or `ndcube.NDCube` objects.
Additional modifications to such packages are outside the scope of ``sunkit-image``. 
We will not copy code from other packages into this one.
```

          generated/gallery/index
          changelog

Mission Statement
=================

> The goal of ``sunkit-image`` is not to be a general purpose image processing library.

Author: ```suggestion
The goal of ``sunkit-image`` is **not** to be a general purpose image processing library.
```

The goal of ``sunkit-image`` is to provide access to image processing routines that are:

1. Focused on being applied to solar image data.

> 2. Are published in the literature or in preparation to be published.

Author: ```suggestion
2. Are published in the literature or in preparation to be published.
   If for any reason, there is doubt to the publication status, the code will only be merged when it's close to actual publication i.e., after approval from the referees.
```

3. Widely used throughout the solar physics community.

>    Examples include co-alignment routines that compensate for incorrect pointing, solar feature identification algorithms, and filtering functions. 

Author: ```suggestion
   Examples include co-alignment routines that compensate for incorrect pointing, solar feature identification algorithms, and filtering functions.
```

> As this is a SunPy affiliated package, the inputs to these routines are required to be Maps (sunpy) or NDCubes (ndcube).

Author: Currently asdf breaks this.

THe time lag one as well, but maybe we should add time series too? :P

Reviewer: What's the input to timelag? I thought it outputted time series?

Author: array-like inputs and outputs

> As this is a SunPy affiliated package, the inputs to these routines are required to be Maps (sunpy) or NDCubes (ndcube).

Author: ```suggestion
As this is a SunPy affiliated package, the inputs to these routines are required to be SunPy objects, Maps or Timeseries from ``sunpy`` or NDCubes from ``ndcube.
This list could expand in the future to other official data structures within the SunPy Ecosystem.
```

Author: Thoughts?

Reviewer: What about high-level objects implemented in the sunpy (or astropy?!) Ecosystem?

Author: What are the high-level objects in the astropy ecosystem?

> As this is a SunPy affiliated package, the inputs to these routines are required to be Maps (sunpy) or NDCubes (ndcube).

Reviewer 2: ```suggestion
```

Author: Do we not want this line?

> If the code is already in a released package, we will wrap calls to the existing package in a way that makes it easy to use with `Map` or `NDCube` objects. Additional modifications to such packages are outside the scope of sunkit-image. 

Author: ```suggestion
If the code is already in a released package, we will wrap calls to the existing package in a way that makes it easy to use with `sunpy.map.Map` or `ndcube.NDCube` objects.
Additional modifications to such packages are outside the scope of sunkit-image. 
```

We will not copy code from other packages into this one.

> 

Author: ```suggestion
```

          generated/gallery/index
          changelog

Mission Statement
=================

> 

Author: ```suggestion
```

The goal of ``sunkit-image`` is **not** to be a general purpose image processing library.

The goal of ``sunkit-image`` is to provide access to image processing routines that are:

1. Focused on being applied to solar image data.
2. Are published in the literature or in preparation to be published.

>    If for any reason, there is doubt to the publication status, the code will only be merged when it's close to actual publication i.e., after approval from the referees.

Author: How do we feel about this wording?
